test/unittests/test_id2iri.py

```diff
@@ -40,16 +40,27 @@ def test_invalid_json_file_name(self) -> None:
             )
 
     def test_replace_id_with_iri(self) -> None:
```

Comment:
I find the name of this test somewhat misleading: the way it stands, I would expect a simple function that does some small replacement - where actually, it tests a top level functionality of the CLI.  
And this closely relates to the modularity issue I mentioned above: This test requires some pretty complex preconditions just to work properly. If all that `id2iri()` does, was to read the two files and pass the file contents to another function, then you could simply test this other function, which would be way easier because it wouldn't require anything in any particular place on the file system. And I would even argue that it would be sufficient to pass a much shorter list of expected IRIs and some short, hand-written XML, just to prove that the replacing works.  
But again: This is an architectural thing and of course does not need to happen in this PR.